vault/external_plugin_test.go

```diff
@@ -0,0 +1,404 @@
+package vault
+
+import (
+	"context"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/hashicorp/vault/helper/namespace"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/helper/consts"
+)
+
+var (
+	compileAuthOnce   sync.Once
+	compileSecretOnce sync.Once
+	pluginBytes       []byte
+)
+
+func testCoreWithPlugin(t *testing.T, typ consts.PluginType) (*Core, string, string) {
+	t.Helper()
+	pluginName, pluginSHA256, pluginDir := compilePlugin(t, typ)
+	conf := &CoreConfig{
+		BuiltinRegistry: NewMockBuiltinRegistry(),
+		PluginDirectory: pluginDir,
+	}
+	core := TestCoreWithSealAndUI(t, conf)
+	core, _, _ = testCoreUnsealed(t, core)
+	return core, pluginName, pluginSHA256
+}
+
+// to mount a plugin, we need a working binary plugin, so we compile one here.
+func compilePlugin(t *testing.T, typ consts.PluginType) (name string, shasum string, pluginDir string) {
+	t.Helper()
+
+	var pluginType, pluginName, builtinDirectory string
+	var once sync.Once
+	switch typ {
+	case consts.PluginTypeCredential:
+		pluginType = "approle"
+		pluginName = "vault-plugin-auth-" + pluginType
+		builtinDirectory = "credential"
+		once = compileAuthOnce
```

Comment:
I don't think this is safe to copy `sync.Once` values. Maybe use `var once *sync.Once`, and copy the references?

Reply:
Oops, good catch, thanks